src/utils.spec.tsx

```diff
@@ -188,4 +206,38 @@ describe(__filename, () => {
       expect(getPathFromQueryString(history)).toEqual(null);
     });
   });
+
+  describe('getCodeLineAnchorGetter', () => {
+    it('returns getCodeLineAnchor if compareInfo is null', () => {
+      const _getCodeLineAnchor = jest.fn();
+      expect(
+        getCodeLineAnchorGetter({ _getCodeLineAnchor, compareInfo: null }),
+      ).toEqual(_getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor if compareInfo is undefined', () => {
+      const _getCodeLineAnchor = jest.fn();
+      expect(
+        getCodeLineAnchorGetter({ _getCodeLineAnchor, compareInfo: undefined }),
+      ).toEqual(_getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor if compareInfo.diff is falsey', () => {
+      const _getCodeLineAnchor = jest.fn();
+      expect(
+        getCodeLineAnchorGetter({
+          _getCodeLineAnchor,
+          compareInfo: { diff: null, mimeType: 'mime/type' },
+        }),
+      ).toEqual(_getCodeLineAnchor);
+    });
+
+    it('returns getCodeLineAnchor from ForwardComparisonMap is compareInfo.diff exists', () => {
+      expect(
+        getCodeLineAnchorGetter({
+          compareInfo: createFakeCompareInfo(),
+        }),
+      ).toEqual(mockGetterFromForwardComparisonMap);
```

Comment:
Could you just inject a stub class, like `getCodeLineAnchorGetter({ _ForwardComparisonMap: ... })`, to solve the problem up above?

Reply:
I tried injecting a mock class, but couldn't get it to work. In any case I think it makes sense to remove all the mocking so that should address most of the comments in this PR.